python-frontend/src/test/java/org/sonar/python/semantic/SymbolTableBuilderTest.java

```diff
@@ -258,12 +264,35 @@ public void print_local_var() {
   public void tuples_in_comp() {
     FunctionDef functionTree = functionTreesByName.get("symbols_in_comp");
     Map<String, Symbol> symbolByName = getSymbolByName(functionTree);
-    assertThat(symbolByName).hasSize(3).containsOnlyKeys("x", "y", "z");
+    assertThat(symbolByName).isEmpty();
+    List<Name> names = getNameFromExpression(((ComprehensionExpression) ((ExpressionStatement) functionTree.body().statements().get(0)).expressions().get(0)).comprehensionFor().loopExpression());
+    assertThat(names).hasSize(3).extracting(Name::name).containsOnly("x", "y", "z");
     for (Symbol symbol : symbolByName.values()) {
       assertThat(symbol.usages()).hasSize(2);
     }
   }
 
+  @Test
+  public void comprehension_scope() {
+    FunctionDef functionTree = functionTreesByName.get("scope_of_comprehension");
+    Map<String, Symbol> symbolByName = getSymbolByName(functionTree);
+    assertThat(symbolByName).containsOnlyKeys("x");
+    assertThat(symbolByName.get("x").usages().stream().map(u -> u.tree().firstToken().line())).hasSize(2).containsExactly(98, 100).doesNotContain(99);
```

Comment:
- I believe that asserting the size is redundant here, as you're already using `containsExactly`.
- Just a minor suggestion: What about using `extracting` like that:
```java
assertThat(symbolByName.get("x").usages()).extracting(u -> u.tree().firstToken().line())
```